java/apirag-server/src/main/java/com/datasqrl/ai/spring/SimpleServer.java

```diff
@@ -0,0 +1,77 @@
+package com.datasqrl.ai.spring;
+
+import com.datasqrl.ai.PlotFunction;
+import com.datasqrl.ai.backend.ChatSession;
+import com.datasqrl.ai.backend.FunctionBackend;
+import com.datasqrl.ai.backend.FunctionDefinition;
+import com.datasqrl.ai.backend.FunctionType;
+import com.datasqrl.ai.backend.ModelBindings;
+import com.datasqrl.ai.backend.RuntimeFunctionDefinition;
+import com.datasqrl.ai.config.ApplicationConfiguration;
+import com.datasqrl.ai.models.ChatClientProvider;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+import java.net.URL;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import lombok.SneakyThrows;
+import okhttp3.internal.concurrent.TaskRunner;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.web.bind.annotation.CrossOrigin;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@SpringBootApplication
+public class SimpleServer {
+
+  public static void main(String[] args) {
+    try {
+      SpringApplication.run(SimpleServer.class, args);
+    } catch (Exception e) {
+      e.printStackTrace();
+    }
+  }
+
+  @CrossOrigin(origins = "*")
+  @RestController
+  public static class MessageController {
+
+    private final FunctionBackend backend;
+    private final String systemPrompt;
+    private final ChatClientProvider<?, ?> chatClientProvider;
+    private final Function<String,Map<String,Object>> getContextFunction;
+
+    @SneakyThrows
+    public MessageController(@Value("${config}") String configFile, @Value("${tools}") String toolsFile) {
+      ApplicationConfiguration configuration = ApplicationConfiguration.fromFile(Path.of(configFile), Path.of(toolsFile));
+      this.getContextFunction = configuration.getContextFunction();
+      this.backend = configuration.getFunctionBackend();
+      this.systemPrompt = configuration.getSystemPrompt();
+      configuration.getPlotFunction().getResourceFile().ifPresent(resourceFile ->
+          configuration.addPlotFunction(backend, SimpleServer.class.getClassLoader().getResource(resourceFile)));
+      this.chatClientProvider = configuration.getChatProviderFactory().create(
+          configuration.getModelConfiguration(), backend, systemPrompt);
```

Comment:
This automatic configuration is blowing my mind 🤯 . A lot to take in, but quite smooth. Do we really need the `addPlotFunction` in the configuration? Couldn't that be done manually here? In a normal use case, users would write the visualization function in their tools file. Not sure you need to handle this as part of the configuration

Reply:
Good point, I am going to pull the plot function stuff into the Configuration class by adding it to the backend directly.

Reply:
I think it would be nice for users to be able to use our default plotting functions. We already spend a fair amount of time fine-tuning them. I expect there will be more of that in the future. For most use cases, you can just use our default data visualization functions without any modifications imho.